reference/enterprise-api-ref/cmdb-api.markdown

```diff
@@ -0,0 +1,317 @@
+
+    
+**Example request (curl):**
+ 
+```
+curl -k --user <username>:<password> \
+  -X GET \
+  https://hub.cfengine.com/api/cmdb?epochFrom=2&hostContextInclude[]=ubuntu \
+  -H 'content-type: application/json' 
+```
+
+**Example response:**
+
+```
+HTTP 200 Ok
+{
+    "data": {
+        "SHA=fdab725e8fb18aa0ad194477be8a2a1338b4b29f6a8597819af89697e432418f": {
+            "classes": {
+                "My_class": {},
+                "My_class2": {
+                    "comment": "comment helps to understand what this class serves for"
+                }
+            },
+            "variables": {
+                "HubCMDB:My.hostname": {
+                    "value": "host1.cfengine.com",
+                    "comment": "comment"
+                },
+                "Namespace:BundleName.VariableName": {
+                    "value": "myvalue"
+                }
+            }
+        }
+    },
+    "meta": {
+        "count": "1",
+        "page": 1,
+        "timestamp": 1619116399,
+        "total": "1",
+        "most_recent_cmdb_change": "13"
+    }
+}
+```
+
+## Get configuration by hostkey
+
+**URI:** https://hub.cfengine.com/cmdb/:hostkey
+
+**Method:** GET
+
+**Parameters:**
+* **hostkey** *(string)*
+  Unique host identifier.
+
+**Example request (curl):**
+
+```
+curl -k --user <username>:<password> \
+  -X GET \
+  https://hub.cfengine.com/api/cmdb/SHA=f622992fa4525070f47da086041a38733496f03a77880f70b1ce6784c38f79ab 
+```
+
+**Example response:**
+
+```
+HTTP 200 Ok
+{
+    "hostkey": "SHA=f622992fa4525070f47da086041a38733496f03a77880f70b1ce6784c38f79ab",
+    "value": {
+        "classes": {
+            "My_class": {},
+            "My_class2": {
+                "comment": ""
+            }
+        },
+        "variables": {
+            "HubCMDB:My.hostname": {
+                "value": "host1.cfengine.com",
+                "comment": "My hostname should be set to this"
+            },
+            "Namespace:BundleName.VariableName": {
+                "value": "myvalue"
+            }
+        }
+    }
+}
+```
+
+## Create new configuration
+
+**URI:** https://hub.cfengine.com/cmdb
+
+**Method:** POST
+
+**Parameters:**
+* **hostkey** *(string)*
+  Unique host identifier.
+* **classes** *(JSON object)*
+  The format is a JSON object where the key is class name and value is another JSON object
+  with an optional `comment` property.
+  Example:
+```
+{
+   "classes":{
+      "My_class":{
+         
+      },
+      "My_class2":{
+         "comment":"comment body"
+      }
+   }
+}
+```
+  
+* **variables** *(JSON object)*
+  The format is a JSON object where the key is variable name and value is another JSON object
+  with a required `value` property and an optional `comment` property.
+  Example:
+```
+{
+   "variables":{
+      "Namespace:BundleName.VariableName":{
+         "value":"myvalue"
```

Comment:
Why are there two separate examples here? Do you really have to specify a json object that has the key "variables" in the value given for the variables parameter? This seems redundant and kind of odd.

Reply:
In the data that the agent loads, yes _variables_ is the key that the definitions must be under.

For modifying data in the cmdb we don't have to re-use this format exactly, but I guess that it's required here since you could set vars and classes in the cmdb concurrently

Reply:
@craigcomstock you still think this is redundant? to have a json illustrating classes and vars separately, with the example showing both?